src/App.tsx

```diff
@@ -106,9 +94,9 @@ export const App: React.FC = () => {
             )}
           >
             <div className="tile is-child box is-success ">
-              {selectedPost && (
-                <PostDetails post={selectedPost} />
-              )}
+              {selectedPost && posts.length ? (
```

Comment:
To avoid `0` rendering use logical operators. `!!` it'll convert the length value to a boolean type
```suggestion
              {selectedPost && !!posts.length ? (
```